core/src/main/scala/com/github/tminglei/slickpg/utils/PgCommonJdbcTypes.scala

```diff
@@ -24,13 +24,10 @@ trait PgCommonJdbcTypes extends JdbcTypesComponent { driver: PostgresDriver =>
       if (r.wasNull) zero else fnFromString(value)
     }
 
-    override def setValue(v: T, p: PreparedStatement, idx: Int): Unit = p.setObject(idx, mkPgObject(v))
+    override def setValue(v: T, p: PreparedStatement, idx: Int): Unit = p.setObject(idx, fnToString(v))
```

Comment:
We should use `PreparedStatement.setObject(parameterIndex, x, targetSqlType)` instead.

_That is, change it to `p.setObject(idx, fnToString(v), java.sql.Types.OTHER)`._